### ISSUE TITLE
Embedder model not using weights learned on Softmax-based losses 

### ISSUE THREAD

**Issue by davidfsemedo** [KevinMusgrave/pytorch-metric-learning#51]
First of all, thank you for this great library!

In Softmax-Based losses, LargeMarginSoftmaxLoss, and in the losses that inherit from this class (SphereFaceLoss, ArcFaceLoss, CosFaceLoss, SphereFaceLoss), a weight matrix self.W is learned during training.

I might be missing something, but it seems that the embedder model does not use these learned weights to project features to the new embedding space.

**Comment by KevinMusgrave** [comment]
@davidfsemedo Yes, this is how it's meant to be. Let's say your embedder model outputs vectors of size 512. If you dataset has 10000 classes, then self.W will transform the 512-dim vectors into 10000-dim vectors, which are then sent to a softmax function. The 512-dim vectors still represent the embedding space. You can verify in the [ArcFace paper](https://arxiv.org/pdf/1801.07698.pdf), near the end of section 3.1: "During testing, we only keep the feature embedding network without the fully connected layer"

You could probably use the 10000-dim vectors, but then the size of your embedding space depends on the number of classes.

**Comment by davidfsemedo** [comment]
@KevinMusgrave Thanks for your quick answer, you're totally right! I the meanwhile I realized it, that's why I closed the issue, even though it was already too late :) Feel free to delete it.
